src/v0/destinations/ga4/customMappingsHandler.js

```diff
@@ -0,0 +1,198 @@
+const get = require('get-value');
+const jsonpath = require('rs-jsonpath');
+const {
+  validateEventName,
+  basicValidation,
+  isReservedEventName,
+  addClientDetails,
+  removeReservedParameterPrefixNames,
+  prepareUserConsents,
+  removeInvalidParams,
+  GA4_RESERVED_PARAMETER_EXCLUSION,
+  getGA4CustomParameters,
+  buildDeliverablePayload,
+  GA4_PARAMETERS_EXCLUSION,
+  prepareUserProperties,
+} = require('./utils');
+const { InstrumentationError } = require('@rudderstack/integrations-lib');
+const {
+  removeUndefinedAndNullRecurse,
+  constructPayload,
+  isDefinedAndNotNull,
+  isEmptyObject,
+  removeUndefinedAndNullValues,
+  isHybridModeEnabled,
+  getIntegrationsObj,
+} = require('../../util');
+const { trackCommonConfig, ConfigCategory, mappingConfig } = require('./config');
+
+const findGA4Events = (eventsMapping, event) => {
+  // Find the event using destructuring and early return
+  const validMappings = eventsMapping.filter(
+    (mapping) =>
+      mapping.rsEventName?.trim().toLowerCase() === event.trim().toLowerCase() &&
+      mapping.ga4EventName,
+  );
+  // Return an empty object if event not found
+  return validMappings;
+};
+
+const handleCustomMappings = (message, Config) => {
+  const { eventsMapping } = Config;
+
+  let rsEvent = get(message, 'event');
```

Comment:
page & group calls will not have `event` in the payload. So basicValidation should be restricted only to track events.
For other message types, eventName should be fetched from `Config.eventsMapping`

Reply:
Addressed